src/triples/batch_random_ot.rs

```diff
@@ -109,10 +185,115 @@ pub async fn batch_random_ot_receiver<C: CSCurve>(
     });
     let out: Vec<_> = stream::iter(tasks).then(|t| t).collect().await;
     let big_k: BitMatrix = out.into_iter().collect();
-
     Ok((delta, big_k.try_into().unwrap()))
 }
 
+pub async fn batch_random_ot_receiver_many<C: CSCurve, const N: usize>(
+    ctx: Context<'_>,
+    mut chan: PrivateChannel,
+) -> Result<Vec<BatchRandomOTOutputReceiver>, ProtocolError> {
+    assert!(N > 0);
+    // Step 3
+    let wait0 = chan.next_waitpoint();
+    let big_y_affine_v: Vec<SerializablePoint<C>> = chan.recv(wait0).await?;
+    
+    let mut big_yv = vec![];
+    let mut deltav = vec![];
+    for i in 0..N {
+        let big_y_affine = big_y_affine_v[i].clone();
+        let big_y = big_y_affine.to_projective();
+        if bool::from(big_y.is_identity()) {
+            return Err(ProtocolError::AssertionFailed(
+                "Big y in batch random OT was zero.".into(),
+            ));
+        }
+    
+        let delta = BitVector::random(&mut OsRng);
+        big_yv.push(big_y);
+        deltav.push(delta);
+    }
+    
+    let big_y_v_arc = Arc::new(big_yv);
+    let big_y_affine_v_arc = Arc::new(big_y_affine_v);
+    
+    // inner is batch, outer is bits
+    let mut choices: Vec<Vec<_>> = Vec::new();
+    for _ in deltav[0].bits() {
+        choices.push(Vec::new());
+    }
+    for j in 0..N {
+        for (i, d_i) in deltav[j].bits().enumerate() {
+            choices[i].push(d_i);
+        }
+    }
+    // wrap in arc
+    let choices: Vec<_> = choices.into_iter().map(|c| Arc::new(c)).collect();
+    
+    let mut tasks = Vec::new();
+    for i in 0..choices.len() {
+        let mut chan = chan.child(i as u64);
+        // clone arcs
+        let d_i_v = choices[i].clone();
+        let big_y_v_arc = big_y_v_arc.clone();
+        let big_y_affine_v_arc= big_y_affine_v_arc.clone();
+        let task = ctx.spawn(async move {
+            let mut x_i_v = Vec::new();
+            let mut big_x_iv = Vec::new();
```

Comment:
I think big_x_iv should be big_x_i_v to maintain consistency with the _v for vector format you've applied elsewhere

Reply:
will keep things consistent